llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1231,11 +1252,22 @@ class VPWidenRecipe : public VPRecipeWithIRFlags, public VPIRMetadata {
       : VPRecipeWithIRFlags(VPDefOpcode, Operands, I), VPIRMetadata(I),
         Opcode(I.getOpcode()) {}
 
+  template <typename IterT>
+  VPWidenRecipe(unsigned VPDefOpcode, unsigned Opcode,
+                iterator_range<IterT> Operands, bool NUW, bool NSW, DebugLoc DL)
```

Comment:
```suggestion
  VPWidenRecipe(unsigned VPDefOpcode, unsigned Opcode,
                ArrayRef<VPValue *> Operands, bool NUW, bool NSW, DebugLoc DL)
```

on current main, using ArrayRef<> for operands should work.